systemtest/src/test/java/io/strimzi/systemtest/migration/MigrationST.java

```diff
@@ -0,0 +1,265 @@
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(),
+            kafka -> kafka.getMetadata().getAnnotations().put(Annotations.ANNO_STRIMZI_IO_KRAFT, "enabled"), testStorage.getNamespaceName());
+
+        controllerPodsSnapshot = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), controllerSelector, 3, controllerPodsSnapshot);
+
+        KafkaUtils.waitUntilKafkaStatusContainsKafkaMetadataState(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaMetadataState.KRaft.name());
+
+        LOGGER.info("6. removing LMFV and IBPV from Kafka config -> brokers and controllers should be rolled");
+        KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getClusterName(), kafka -> {
+            kafka.getSpec().getKafka().getConfig().remove("log.message.format.version");
+            kafka.getSpec().getKafka().getConfig().remove("inter.broker.protocol.version");
+        }, testStorage.getNamespaceName());
+
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), brokerSelector, 3, brokerPodsSnapshot);
+        RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(testStorage.getNamespaceName(), controllerSelector, 3, controllerPodsSnapshot);
+
+        LOGGER.info("7. creating KafkaTopic: {} and checking if the metadata are only in KRaft", kraftTopicName);
+        resourceManager.createResourceWithWait(extensionContext, KafkaTopicTemplates.topic(testStorage.getClusterName(), kraftTopicName, testStorage.getNamespaceName()).build());
+
+        LOGGER.info("7a. checking if metadata about KafkaTopic: {} are in KRaft controller", kraftTopicName);
+
+        assertThatTopicIsPresentInKRaftMetadata(controllerSelector, kraftTopicName);
+
+        LOGGER.info("7b. checking if we are able to do a message transmission on KafkaTopic: {}", kraftTopicName);
+
+        immediateClients = new KafkaClientsBuilder(immediateClients)
+            .withTopicName(kraftTopicName)
+            .build();
+
+        resourceManager.createResourceWithWait(extensionContext,
+            immediateClients.producerTlsStrimzi(testStorage.getClusterName()),
+            immediateClients.consumerTlsStrimzi(testStorage.getClusterName())
+        );
+
+        ClientUtils.waitForClientsSuccess(testStorage.getProducerName(), testStorage.getConsumerName(), testStorage.getNamespaceName(), testStorage.getMessageCount());
+        // TODO: continuous clients are failing due to missing messages (one or two) during the whole process. This happens from time to time in upgrade/downgrade tests as well
+        // it needs to be fixed on test-clients side
+        //  LOGGER.info("8. migration is completed, waiting for continuous clients to finish");
+        //  ClientUtils.waitForClientsSuccess(continuousProducerName, continuousConsumerName, testStorage.getNamespaceName(), continuousMessageCount);
+    }
```

Comment:
Assuming migration went as expected, do we expect Zookeeper `PodSet` and its `Pods` to be removed automatically ?

Reply:
ZooKeeper removal is not in this PR yet. It's in the oven ... I am cooking it and will be here soon ;-)